x/mongo/driver/operation/listDatabases.go

```diff
@@ -62,76 +60,64 @@ func buildListDatabasesResult(response bsoncore.Document, srvr driver.Server) (L
 	ir := ListDatabasesResult{}
 	for _, element := range elements {
 		switch element.Key() {
-
 		case "totalSize":
 			var ok bool
 			ir.TotalSize, ok = element.Value().AsInt64OK()
 			if !ok {
-				err = fmt.Errorf("response field 'totalSize' is type int64, but received BSON type %s: %s", element.Value().Type, element.Value())
+				return ir, fmt.Errorf("response field 'totalSize' is type int64, but received BSON type %s: %s", element.Value().Type, element.Value())
 			}
-
 		case "databases":
-			// TODO: Make operationgen handle array results.
 			arr, ok := element.Value().ArrayOK()
 			if !ok {
-				err = fmt.Errorf("response field 'databases' is type array, but received BSON type %s", element.Value().Type)
-				continue
+				return ir, fmt.Errorf("response field 'databases' is type array, but received BSON type %s", element.Value().Type)
 			}
 
 			var tmp bsoncore.Document
 			marshalErr := bson.Unmarshal(arr, &tmp)
```

Comment:
Optional: Now that we're not trying to keep track of the `err` variable value, consider renaming this variable to `err`.

Reply:
Done here and elsewhere in the file.